File: app/src/main/java/com/fstyle/structure_android/data/source/remote/api/service/NameApi.java
package com.fstyle.structure_android.data.source.remote.api.service;

import com.fstyle.structure_android.data.model.User;
import com.fstyle.structure_android.data.model.UsersList;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by le.quang.dao on 10/03/2017.
 */

public interface NameApi {
    @GET("/search/users")
    Observable<UsersList> searchGithubUsers(@Query("per_page") int limit,
            @Query("q") String searchTerm);

    @GET("/users/{username}")
    Observable<User> getUser(@Path("username") String username);
}

File: app/src/main/java/com/fstyle/structure_android/data/source/remote/api/service/NameApi.kt
package com.fstyle.structure_android.data.source.remote.api.service

import com.fstyle.structure_android.data.model.User
import com.fstyle.structure_android.data.source.remote.api.response.SearchUserResponse
import io.reactivex.Observable
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Created by le.quang.dao on 10/03/2017.
 */

interface NameApi {
  @GET("/search/users")
  fun searchGithubUsers(@Query("per_page") limit: Int,
      @Query("q") searchTerm: String?): Single<SearchUserResponse>

  @GET("/users/{username}")
  fun getUser(@Path("username") username: String?): Single<User>
}
